test/kfac_cases.py

```diff
@@ -128,3 +128,41 @@
             "device": device,
         }
         KFAC_EXACT_ONE_DATUM_CASES.append(case_with_device)
+
+
+# Add test cases here, devices and loss function with different reductions will be
+# added automatically below
+SINGLE_LAYER_CASES_NO_DEVICE_NO_LOSS_FUNC = [
+    ###############################################################################
+    #                                  REGRESSION                                 #
+    ###############################################################################
+    # single-layer linear network with scalar output
+    {
+        "model_func": lambda: Linear(6, 1),
+        "data": lambda: [
+            (rand(5, 6), regression_targets((5, 1))),
+            (rand(5, 6), regression_targets((5, 1))),
+        ],
+        "seed": 0,
+    },
+    # single-layer linear network with vector output
+    {
+        "model_func": lambda: Linear(5, 3),
+        "data": lambda: [
+            (rand(7, 5), regression_targets((7, 3))),
+            (rand(7, 5), regression_targets((7, 3))),
+        ],
+        "seed": 0,
+    },
```

Comment:
Could add a `Sequential(Conv2d, Flatten)` which also satisfies the test (I think!? Might have to double-check tomorrow as it is late already here).

Reply:
I added a test for single-layer models with weight sharing (`Linear` and `Conv2d`). You can double check if the math makes sense to you (note the assumption of a mean reduction function for the reduce setting).